xdsl/dialects/builtin.py

```diff
@@ -408,6 +410,7 @@ class DictionaryAttr(GenericData[dict[str, Attribute]]):
     def parse_parameter(parser: BaseParser) -> dict[str, Attribute]:
         # force MLIR style parsing of attribute
         from xdsl.parser import MLIRParser
+        assert isinstance(parser, MLIRParser)
```

Comment:
Mh, that reads hacky! Doesn't it crash when parsing xDSL syntax?

Reply:
Actually, neither of the parser are actually parsing this, right?
Should we return an exception instead, since this code is never executed anyway?

Reply:
Fixed in the other PR!